src/libraries/Common/tests/System/Security/Cryptography/SignatureSupport.cs

```diff
@@ -1,51 +1,69 @@
+                if (!OperatingSystem.IsLinux())
+                {
+                    return true;
+                }
 
-            switch (algorithm)
+                switch (algorithm)
+                {
+                    case ECDsa ecdsa:
+                        try
+                        {
+                            ecdsa.SignData(Array.Empty<byte>(), HashAlgorithmName.SHA1);
+                            return true;
+                        }
+                        catch (CryptographicException)
+                        {
+                            return false;
+                        }
+                    case RSA rsa:
+                        try
+                        {
+                            rsa.SignData(Array.Empty<byte>(), HashAlgorithmName.SHA1, RSASignaturePadding.Pkcs1);
+                            return true;
+                        }
+                        catch (CryptographicException)
+                        {
+                            return false;
+                        }
+                    default:
+                        throw new NotSupportedException($"Algorithm type {algorithm.GetType()} is not supported.");
+                }
+            }
+            finally
             {
-                case ECDsa ecdsa:
-                    try
-                    {
-                        ecdsa.SignData(Array.Empty<byte>(), HashAlgorithmName.SHA1);
-                        return true;
-                    }
-                    catch (CryptographicException)
-                    {
-                        return false;
-                    }
-                    finally
-                    {
-                        algorithm.Dispose();
-                    }
-                case RSA rsa:
-                    try
-                    {
-                        rsa.SignData(Array.Empty<byte>(), HashAlgorithmName.SHA1, RSASignaturePadding.Pkcs1);
-                        return true;
-                    }
-                    catch (CryptographicException)
-                    {
-                        return false;
-                    }
-                    finally
-                    {
-                        algorithm.Dispose();
-                    }
-                default:
-                    throw new NotSupportedException($"Algorithm type {algorithm.GetType()} is not supported.");
+                algorithm.Dispose();
             }
+#endif
         }
+
+        public static bool SupportsRsaSha1Signatures
+        {
+            get
+            {
+                if (_supportsRsaSha1Signatures == 0)
+                {
+                    bool supported = CanProduceSha1Signature(new RSACryptoServiceProvider());
+                    _supportsRsaSha1Signatures = supported ? 1 : -1;
+                }
+                return _supportsRsaSha1Signatures == 1;
+            }
+        }  
```

Comment:
My preference would be for this to be as localized as possible to S.S.C.Xml. An implementation should normally be passed in because their behavior might be different, and we support multiple providers on a single platform. For example, `RSAOpenSsl` and `RSA.Create()` on macOS use two entirely different RSA implementations.

So tests should be written with that in mind. Since S.S.C.Xml usually uses `RSACryptoServiceProvider` internally to do its RSA-y stuff, this makes sense there. But I wouldn't want this to be a defact-o way to determine "Is RSA supported or not".

tl;dr: I think this should be in `S.S.C.Xml.Tests` and renamed to `SupportsRsaSha1CspSignatures` or something.

Reply:
I think @bartonjs suggested to use the same file for both: https://github.com/dotnet/runtime/pull/69713#discussion_r880826222.

Reply:
I neglected to pay attention to the fact that the shared file requires an implementation to be passed in when I made my comment.

I agree with Kevin's comments here and withdraw my prior one.